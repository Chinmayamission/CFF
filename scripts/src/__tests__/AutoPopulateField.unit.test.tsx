import React from "react";
import { shallow, mount, render } from "enzyme";
import CustomForm from "../form/CustomForm";
import { getSchemaFromResults } from "../form/form_widgets/AutoPopulateField";

declare global {
  namespace NodeJS {
    interface Global {
      document: Document;
      window: Window;
      navigator: Navigator;
      fetch: any;
    }
  }
}

// afterEach(() => {
//     if (global.fetch) {
//         global.fetch.mockClear();
//         delete global.fetch;
//     }
// });

// function mockFetch(mockSuccessResponse) {
//     const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
//     const mockFetchPromise = Promise.resolve({ // 3
//         json: () => mockJsonPromise,
//     });
//     global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
// }

describe("getSchemaFromResults", () => {
  it("base case", () => {
    let results = [
      {
        address1: "A",
        city: "City",
        state: "State",
        zip: "Zipcode",
        name: "Center Name 1"
      }
    ];
    let schema = {
      type: "object",
      properties: {
        name: { type: "string" },
        address1: { type: "string" },
        state: { type: "string" },
        zip: { type: "string" }
      }
    };
    let titleAccessor = "name";
    let newSchema = getSchemaFromResults({ results, schema, titleAccessor });
    expect(newSchema).toMatchSnapshot();
  });
  it("should preserve title from schema", () => {
    let results = [
      {
        address1: "A",
        city: "City",
        state: "State",
        zip: "Zipcode",
        name: "Center Name 1"
      }
    ];
    let schema = {
      type: "object",
      properties: {
        name: { type: "string" },
        address1: { type: "string" },
        state: { type: "string" },
        zip: { title: "Zip / Pincode", type: "string" }
      }
    };
    let titleAccessor = "name";
    let newSchema = getSchemaFromResults({ results, schema, titleAccessor });
    expect(newSchema).toMatchSnapshot();
    expect(newSchema.dependencies.name.oneOf[0].properties.zip.title).toEqual(
      "Zip / Pincode"
    );
  });
});
