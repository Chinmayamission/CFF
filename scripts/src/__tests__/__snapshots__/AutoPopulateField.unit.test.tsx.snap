// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSchemaFromResults base case 1`] = `
Object {
  "dependencies": Object {
    "name": Object {
      "oneOf": Array [
        Object {
          "properties": Object {
            "address1": Object {
              "default": "A",
              "enum": Array [
                "A",
              ],
              "readOnly": true,
              "type": "string",
            },
            "name": Object {
              "const": "Center Name 1",
            },
            "state": Object {
              "default": "State",
              "enum": Array [
                "State",
              ],
              "readOnly": true,
              "type": "string",
            },
            "zip": Object {
              "default": "Zipcode",
              "enum": Array [
                "Zipcode",
              ],
              "readOnly": true,
              "type": "string",
            },
          },
        },
      ],
    },
  },
  "properties": Object {
    "name": Object {
      "enum": Array [
        "Center Name 1",
      ],
      "type": "string",
    },
  },
  "type": "object",
}
`;
